src/box/lua/connpool.lua

```diff
@@ -71,24 +78,49 @@ local function connect(instance_name, opts)
     return conn
 end
 
+local function is_candidate_connected(candidate)
+    local conn = connections[candidate]
+    return conn and conn.state == 'active' and conn:mode() ~= nil
+end
+
+-- Checks whether the candidate instance is not in some of the
+-- connecting/fetching state.
+local function is_candidate_checked(candidate)
+    local conn = connections[candidate]
+
+    return not conn or
+           is_candidate_connected(candidate) or
+           conn.state == 'error' or
+           conn.state == 'closed'
+end
+
 local function connect_to_candidates(candidates)
     local delay = WATCHER_DELAY
-    local conn_opts = {connect_timeout = WATCHER_TIMEOUT}
-    local connected_candidates = {}
+    local time_connect_end = clock.monotonic() + WATCHER_TIMEOUT
```

Comment:
nit: i like a name like `deadline` better

Reply:
Done. Thank you. For some reason I've chosen some strange name.

Reply:
Seems fixed now.